### PR TITLE
Task-36728: Not display externals on people directory and contact suggestions.

### PR DESCRIPTION
**Pull request by ahmadmido**

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/rest/api/RestUtils.java [hunk 1]
```diff
@@ -124,6 +126,15 @@ private static boolean isSupportedFormat(String expectedFormat, String[] support
...
     return ConversationState.getCurrent().getIdentity().isMemberOf(ADMIN_GROUP);
   }
+
+  /**
+   * Check if the user is a member of the external group
+   *
+   * @return
+   */
+  public static boolean isMemberOfExternalGroup(String userId) throws Exception{
+    return getIdentity(userId).isMemberOf(EXTERNAL_GROUP);
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/RestUtils.java]:
```suggestion
OrganizationService organizationService = CommonsUtils.getService(OrganizationService.class);
    return organizationService.getMembershipHandler().findMembershipsByUserAndGroup(userId, EXTERNAL_GROUP).size() > 0;
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/RestUtils.java [hunk 2]
```diff
@@ -258,4 +269,20 @@ public static IdentityManager getIdentityManager() {
     }
     return identityManager;
   }
+
+  private static org.exoplatform.services.security.Identity getIdentity(String userId) throws Exception {
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/RestUtils.java]: Not needed

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — CHANGES_REQUESTED** [review]
We need to change the query by adding like below:

SELECT distinct identity.* FROM plf6.soc_identities as identity inner join plf6.soc_identity_properties as properties on identity.identity_id = properties.identity_id and not exists (select properties_tmp.IDENTITY_ID from plf6.soc_identity_properties as properties_tmp where  properties_tmp.IDENTITY_ID = identity.IDENTITY_ID and  properties_tmp.name = "external" and  properties_tmp.value="true");

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 3]
```diff
@@ -310,7 +310,7 @@ public Response getUsers(@Context UriInfo uriInfo,
     List<DataEntity> profileInfos = new ArrayList<DataEntity>();
     for (Identity identity : identities) {
       ProfileEntity profileInfo = EntityBuilder.buildEntityProfile(identity.getProfile(), uriInfo.getPath(), expand);
-      //
+      if(RestUtils.isMemberOfExternalGroup(identity.getRemoteId())) continue;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: No needed after changing the query

File: component/service/src/main/java/org/exoplatform/social/rest/suggest/PeopleRestServices.java [hunk 4]
```diff
@@ -255,7 +256,7 @@ public Response getSuggestions(@Context SecurityContext sc, @Context UriInfo uri
       }
       for (Entry<Identity, Integer> suggestion : suggestions.entrySet()) {
         Identity id = suggestion.getKey();
-
+        if(RestUtils.isMemberOfExternalGroup(id.getRemoteId())) continue;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/suggest/PeopleRestServices.java:259]: you should check directly id.getProfile.getProperty("external")

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java [hunk 5]
```diff
@@ -710,7 +710,12 @@ public int getIdentitiesByProfileFilterCount(String providerId, ProfileFilter pr
     ListAccess<IdentityEntity> list = getIdentityDAO().findIdentities(xFilter);
 
     try {
-      return list.getSize();
+      int listCount = 0;
+      for(IdentityEntity identityEntity : list.load(-1, -1)){
```
Comment (ahamdi) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java:723]: This is a performance killer, please rework it and avoid FOR loops.

**Review by ahamdi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**